R/LSBuffonsneedlesimulation.R

```diff
@@ -0,0 +1,279 @@
+#
+# Copyright (C) 2019 University of Amsterdam
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+LSBuffonsneedlesimulation<- function(jaspResults, dataset, options, state = NULL){
+
+  
+  # simulation results
+  #simulateResults <- simulate_throws(options[["n"]], options[["length"]])
+  
+  
+  if(is.null(jaspResults[["simulateResults"]])){ #test whether the state is empty
+    
+    # if empty, create a new state
+    simulateResults <- createJaspState()
+    simulateResults$dependOn(c("n", "length")) #, "a", "b", "options[["length"]]", "options[["CI"]]"))
+    jaspResults[["simulateResults"]] <- simulateResults
+    
+    jaspResults[["simulateResults"]]$object <- simulateThrows(options[["n"]], options[["length"]])  
+    
+  }
+  # if not, retrieve the values
+  crosses <- jaspResults[["simulateResults"]][["object"]][["k"]]
+  xs <- jaspResults[["simulateResults"]][["object"]][["xs"]] 
+  xe <- jaspResults[["simulateResults"]][["object"]][["xe"]] 
+  ys <- jaspResults[["simulateResults"]][["object"]][["ys"]] 
+  ye <- jaspResults[["simulateResults"]][["object"]][["ye"]]
+  
+  xsCrosses <- jaspResults[["simulateResults"]][["object"]][["xsCrosses"]] 
+  xeCrosses <- jaspResults[["simulateResults"]][["object"]][["xeCrosses"]] 
+  ysCrosses <- jaspResults[["simulateResults"]][["object"]][["ysCrosses"]] 
+  yeCrosses <- jaspResults[["simulateResults"]][["object"]][["yeCrosses"]]  
```

Comment:
given that you now pass the entire jaspResults into the output functions, you do not need to extract these values here, right?

Reply:
Yeah but the code would then be extra long and redundant. If that's ok I can remove the steps of extracting values.

Reply:
I am not sure whether I follow... the highlighted code does not do anything else than extract the values which you don't use in the current function, right? Then inside the other functions you already extract these values already. The point is that this piece of code currently does not serve any purpose, really.

Reply:
Got it! I have removed these values here.